src/main/kotlin/com/jayrave/moshi/pristineModels/Mapper.kt

```diff
@@ -90,7 +88,7 @@ abstract class Mapper<T : Any> {
             return create(object : Value<T> {
                 override fun <F> of(fieldMapping: FieldMapping<T, F>): F {
                     return when (fieldMapping) {
-                        is FieldMappingImpl -> fieldMapping.lastReadValueInCurrentThread()
+                        is FieldMappingImpl -> mapValues[fieldMapping.name] as F
```

Comment:
I think holding on to it in a map here & the subsequent casting should be ok since `fieldMapping.read` is guaranteed to return the correct type in the first place. This also make sure that the `JsonAdapterForMapper` in each `Mapper` class uses only the local state from inside this method & so doesn't step on any nested calls' state 👍